### PR TITLE
SONARJAVA-1441 Remove SourceFile JavaAstScanner.scanSingleFile(File file, VisitorsBridge visitorsBridge)

### PR DESCRIPTION
**Pull request by mpaladin**

### CODE REVIEW

File: java-squid/src/main/java/org/sonar/java/ast/visitors/VisitorContext.java [hunk 1]
```diff
@@ -19,49 +19,19 @@
...
-  }
-
-  private void addSourceCode(SourceCode child) {
-    peekSourceCode().addChild(child);
-    sourceCodeStack.push(child);
-  }
-
-  public SourceCode peekSourceCode() {
-    return sourceCodeStack.peek();
+  public VisitorContext() {
```
Comment (SonarTech) [inline java-squid/src/main/java/org/sonar/java/ast/visitors/VisitorContext.java]: MAJOR Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1186)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue:
* MAJOR 1 major

Watch the comments in this conversation to review them.

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue:
* MAJOR 1 major

Watch the comments in this conversation to review them.

File: java-squid/src/main/java/org/sonar/java/ast/JavaAstScanner.java [hunk 2]
```diff
@@ -69,16 +55,10 @@ public JavaAstScanner(ActionParser<Tree> parser) {
    */
   public JavaAstScanner(JavaAstScanner astScanner) {
     this.parser = astScanner.parser;
-    this.index = astScanner.index;
   }
 
   public void scan(Iterable<File> files) {
```
Comment (benzonico) [inline java-squid/src/main/java/org/sonar/java/ast/JavaAstScanner.java]: Given the result of that method I guess that either this one or simpleScan should go away...

File: java-squid/src/main/java/org/sonar/java/model/VisitorsBridgeForTests.java [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Original VisitorsBridge renamed to InternalVisitorsBridge because VisitorsBridge
+ * is the recommended way for custom rules.
+ * We need to customize VisitorsBridge to keep compatibility with CheckMessageVerifier.
+ * InternalVisitorsBridge can be refactored once CheckMessageVerifier support will be dropped.
+ */
```
Comment (benzonico) [inline java-squid/src/main/java/org/sonar/java/model/VisitorsBridgeForTests.java]: This javaDoc seems wrong.

**Pull request merged**